scipy/optimize/linesearch.py

```diff
@@ -684,7 +683,7 @@ def scalar_search_armijo(phi, phi0, derphi0, c1=1e-4, alpha0=1, amin=0):
 
     # Otherwise compute the minimizer of a quadratic interpolant:
 
-    alpha1 = -(derphi0) * alpha0**2 / 2.0 / (phi_a0 - phi0 - derphi0 * alpha0)
+    alpha1 = _quadmin(0, phi0, derphi0, alpha0, phi_a0)
```

Comment:
it's not obvious that this is the same as the old code, with the `return None` that `_quadmin` can do. Can you explain this change a bit?

Reply:
I add code to handle this case.

Reply:
okay, verified that this matches:
```
import numpy as np


def _quadmin(a, fa, fpa, b, fb):
    """
    Finds the minimizer for a quadratic polynomial that goes through
    the points (a,fa), (b,fb) with derivative at a of fpa,

    """
    # f(x) = B*(x-a)^2 + C*(x-a) + D
    with np.errstate(divide='raise', over='raise', invalid='raise'):
        try:
            D = fa
            C = fpa
            db = b - a * 1.0
            B = (fb - D - C * db) / (db * db)
            xmin = a - C / (2.0 * B)
        except ArithmeticError:
            return None
    if not np.isfinite(xmin):
        return None
    return xmin


for _ in range(100):
    derphi0 = np.random.rand(1)
    alpha0 = np.random.rand(1)
    phi_a0 = np.random.rand(1)
    phi0 = np.random.rand(1)

    alpha1 = -(derphi0) * alpha0**2 / 2.0 / (phi_a0 - phi0 - derphi0 * alpha0)
    alpha2 = _quadmin(0, phi0, derphi0, alpha0, phi_a0)
    np.testing.assert_allclose(alpha1, alpha2, atol=0, rtol=5e-16)
```